scapy/contrib/pim.py

```diff
@@ -0,0 +1,252 @@
+# This file is part of Scapy
+# Scapy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# any later version.
+#
+# Scapy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Scapy. If not, see <http://www.gnu.org/licenses/>.
+#
+# scapy.contrib.description = Protocol Independent Multicast (PIM)
+# scapy.contrib.status = loads
+"""
+References:
+  - https://tools.ietf.org/html/rfc4601
+  - https://www.iana.org/assignments/pim-parameters/pim-parameters.xhtml#pim-parameters-1  # noqa: E501
+"""
+from scapy.packet import Packet, bind_layers
+from scapy.fields import BitFieldLenField, BitField, BitEnumField, ByteField, \
+    ShortField, XShortField, IPField, PacketListField, \
+    IntField, FieldLenField, BoundStrLenField, FlagsField
+from scapy.layers.inet import IP
+from scapy.utils import checksum
+from scapy.compat import chb, orb
+from scapy.config import conf
+from scapy.volatile import RandInt
+
+
+PIM_TYPE = {
+    0: "Hello",
+    1: "Register",
+    2: "Register-Stop",
+    3: "Join/Prune",
+    4: "Bootstrap",
+    5: "Assert",
+    6: "Graft",
+    7: "Graft-Ack",
+    8: "Candidate-RP-Advertisement"
+}
+
+
+class PIMv2Hdr(Packet):
+    name = "Protocol Independent Multicast Version 2 Header"
+    fields_desc = [BitField("version", 2, 4),
+                   BitEnumField("type", 0, 4, PIM_TYPE),
+                   ByteField("reserved", 0),
+                   XShortField("chksum", None)]
+
+    def post_build(self, p, pay):
+        """
+        Called implicitly before a packet is sent to compute and place PIM checksum.  # noqa: E501
+
+        Parameters:
+          self    The instantiation of an PIMv2Hdr class
+          p       The PIMv2Hdr message in hex in network byte order
+          pay     Additional payload for the PIMv2Hdr message
+        """
+        p += pay
+        if self.chksum is None:
+            ck = checksum(p)
+            p = p[:2] + chb(ck >> 8) + chb(ck & 0xff) + p[4:]
+        return p
+
+
+def _guess_pim_tlv_class(h_classes, defaultname, pkt, **kargs):
+    cls = conf.raw_layer
+    if len(pkt) >= 2:
+        tlvtype = orb(pkt[1])
+        clsname = h_classes.get(tlvtype, defaultname)
+        cls = globals()[clsname]
```

Comment:
Could you try to avoid using the `globals()` and use the class names directly ? I know we use it elsewhere but it's not a great practice

Reply:
I apply feedback.
I changed it like below link.
 - https://github.com/secdev/scapy/pull/2757/commits/d6c479a90ec66bea2a13ecba0af60b6cae8a09f4

Reply:
@gpotter2 
I applied the feedback and tested it, but it failed.
This is a problem related to `npcap_oem_key`.
but I have not found the cause of this problem.
Could it have something to do with the pull request code my posted?

```
Build started
git clone -q https://github.com/secdev/scapy.git C:\projects\scapy
git fetch -q origin +refs/pull/2757/merge:
git checkout -qf FETCH_HEAD
Running Install scripts
.\.config\appveyor\InstallNpcap.ps1
Get-ChildItem : Cannot find path 'npcap_oem_key' because it does not exist.
At C:\projects\scapy\.config\appveyor\InstallNpcap.ps1:15 char:21
+     $user, $pass = (Get-ChildItem Env:npcap_oem_key).Value.replace("` ...
+                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (npcap_oem_key:String) [Get-ChildItem], ItemNotFoundException
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetChildItemCommand
 
Checksums matches !
Installing:
C:\projects\scapy\.config\appveyor\npcap-0.96.exe
Npcap installation completed
.\.config\appveyor\InstallWindumpNpcap.ps1
Checksums matches !
    Directory: C:\Program Files
Mode                LastWriteTime         Length Name                                                                                                                                                                                
----                -------------         ------ ----                                                                                                                                                                                
d-----        8/12/2020   7:03 PM                Windump     
``` 

 - https://ci.appveyor.com/project/secdev/scapy/builds/34625314/job/9f0ut31p164wesso